config/ConfigParse.cpp

```diff
@@ -0,0 +1,229 @@
+{
+	std::vector<std::pair<std::string, int> > configs;
+	for (int i = 0; i < str.size(); i ++)
+	{
+		int start = i;
+		for (start; str[start] == '\t'; start ++)
+			;
+		int	indent = start - i;
+		i = start;
+		int	end = i;
+		for (end; str[end] != '\n'; end ++)
+			;
+		i = end;
+		if (start != end)
+			configs.push_back(std::make_pair(str.substr(start, end - start), indent));
+	}
+	return configs;
+}
+
+std::vector<std::vector<std::pair<std::string, int> > > findChunck(std::vector<std::pair<std::string, int> > indents)
+{
+	std::vector<std::vector<std::pair<std::string, int> > > multiChunck;
+	int start = 0;
+	int	end = 0;
+	
+	for (int i = 0; i < indents.size(); i ++)
+	{
+		start = i;
+		for (i; i < indents.size() && indents[i].first != "server"; i ++)
+			;
+		end = i;
+		if (start != end)
+		{
+			std::vector<std::pair<std::string, int> > chunck(indents.begin() + start, indents.begin() + end);
+			multiChunck.push_back(chunck);
+		}
+	}
+	return multiChunck;
+}
+
+std::string	parseString(const std::vector<std::pair<std::string, int> > chunck, std::string keyword)
+{
+	for (int i = 0; i < chunck.size(); i ++)
+	{
+		if (chunck[i].first.find(keyword) != std::string::npos)
+		{
+			std::string str = chunck[i].first;
+			std::string::iterator start = str.begin() + keyword.size() + 1;
+			while(start != str.end() && std::isspace(*start))
+				++ start;
+			std::string::iterator end = str.end();
+			while (end != start && std::isspace(*end))
+				-- end;
+			return (std::string(start, end + 1));
+		}
+	}
+	return "";
+}
+
+std::vector<std::string>	parseMethod(const std::vector<std::pair<std::string, int> > chunck)
+{
+	std::vector<std::string> methods;
+	std::string str;
+
+	str = parseString(chunck, "httpMethods");
+	if (str.empty())
+		return std::vector<std::string>();
+	std::istringstream iss(str);
+	std::string method;
+
+	while (iss >> method)
+		methods.push_back(method);
+	return methods;
+}
+
+std::vector<std::string>	parsePath(const std::vector<std::pair<std::string, int> > chunck)
+{
+	std::vector<std::string> paths;
+	std::string				keyword = "path";
+	int	i;
+
+	for (i = 0; i < chunck.size() && chunck[i].first.find(keyword) == std::string::npos; i ++)
+		;
+	int	indent = chunck[i].second;
+	i ++;
+	for (i; i < chunck.size() && chunck[i].second == indent + 1; i++)
+	{
+		std::istringstream iss(chunck[i].first);
+		std::string path;
+		iss >> path;
+		paths.push_back(path);
+	}
+	return paths;
+}
+
+std::map<std::string, std::string> parseCgi(std::vector<std::pair<std::string, int> > chunck)
+{
+	std::map<std::string, std::string> cgi;
+	std::string keyword = "cgi";
+	int	i;
+
+	for (i = 0; i < chunck.size() && chunck[i].first.find(keyword) == std::string::npos; i ++)
+		;
+	int	indent = chunck[i].second;
+	i ++;
+	for (i; i < chunck.size() && chunck[i].second == indent + 1; i ++)
+	{
+		std::istringstream iss(chunck[i].first);
+		std::string extension;
+		std::string path;
+		iss >> extension >> path;
```

Comment:
are extension/paths that contain whitespaces allowed?

Reply:
No, and it would be fine to consider as a grammar error if the path/extension has a whitespace